gleam_stdlib/src/list.gleam

```diff
@@ -4,6 +4,9 @@ pub enum Empty =
 pub enum NotFound =
   | NotFound
 
+pub enum ListLengthMismatch =
+  | ListLengthMismatch
```

Comment:
Tiny nitpick- can we call this `LengthMismatch`? Thank you.

Reply:
sure!